src/render/painter.js

```diff
@@ -287,26 +287,9 @@ class Painter {
      * new tiles at the same location, while retaining previously drawn pixels.
      */
     clearStencil() {
-        const context = this.context;
-        const gl = context.gl;
-
         this.nextStencilID = 1;
         this.currentStencilSource = undefined;
-
-        // As a temporary workaround for https://github.com/mapbox/mapbox-gl-js/issues/5490,
-        // pending an upstream fix, we draw a fullscreen stencil=0 clipping mask here,
-        // effectively clearing the stencil buffer: once an upstream patch lands, remove
-        // this function in favor of context.clear({ stencil: 0x0 })
```

Comment:
Are you sure this is not still an issue affecting older gpus?

Reply:
I provided a build to users that reported experiencing this issue to test. Pending a confirmation check in https://github.com/mapbox/mapbox-gl-js/issues/5490#issuecomment-878624251 I can't be 100% sure that this would be safe to remove. If I get positive confirmation, I'll create a separate PR so we can at least include this simplfiication in the next release.